### ISSUE TITLE
[YSQL] Batch nested loops closing connection from Hasura query due to invalid eqop in hash clause

### ISSUE THREAD

**Issue by mislam-yb** [yugabyte/yugabyte-db#16105]
Jira Link: [DB-5510](https://yugabyte.atlassian.net/browse/DB-5510)
### Description

When using batched nested loops the following query causes the connection to disconnect:

```
SELECT c.column_name, c.is_nullable = 'YES', c.udt_name, c.character_maximum_length, c.numeric_precision,
	c.numeric_precision_radix, c.numeric_scale, c.datetime_precision, 8 * typlen, c.column_default, pd.description,
	c.identity_increment
	FROM information_schema.columns AS c
	JOIN pg_type AS pgt ON c.udt_name = pgt.typname
	LEFT JOIN pg_catalog.pg_description as pd ON pd.objsubid = c.ordinal_position AND pd.objoid = (
	  SELECT oid FROM pg_catalog.pg_class
	  WHERE relname = c.table_name AND relnamespace = (
	    SELECT oid FROM pg_catalog.pg_namespace
	    WHERE nspname = c.table_schema
	    )
	  )
	where table_catalog = 'yugabyte' AND table_schema = 'clusters' AND table_name = 'clusters';
```

The postgres logs show
```
TRAP: FailedAssertion("!(((_Bool) ((eqop) != ((Oid) 0))))", File: "../../../../../../src/postgres/src/backend/executor/nodeYbBatchedNestloop.c", Line: 356)
```

This error occured for the latest debug build for 2.17.2.0.

[DB-5510]: https://yugabyte.atlassian.net/browse/DB-5510?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by tanujnay112** [comment]
Keeping this open as original fix was wrong.

**Comment by tanujnay112** [comment]
This is fixed by d494baf9aec8bf4642b3ced7c02db655b01a62d0 now.
